$NetBSD: patch-al,v 1.9 2013/05/12 14:41:14 makoto Exp $

Thu Sep  9 09:03:30 UTC 2010
The configure script requests sqlite3 compiled with SECURE_DELETE enabled
by default. Enabling it be default would mean a performance penalty for
other applications.

Instead override the configure script and in patch-al do
"PRAGMA secure_delete = 1;" when opening the sqlite session.

--- storage/src/mozStorageConnection.cpp.orig	2013-01-04 23:44:44.000000000 +0000
+++ storage/src/mozStorageConnection.cpp
@@ -584,6 +584,11 @@ Connection::initialize(nsIFile *aDatabas
       break;
   }
 
+  // XXX tnn: the configure script demands that sqlite3 is compiled with
+  // SECURE_DELETE on by default. sqlite3 in pkgsrc does not have that,
+  // so instead we enable secure_delete manually here.
+  (void)ExecuteSimpleSQL(NS_LITERAL_CSTRING("PRAGMA secure_delete = 1;"));
+
   return NS_OK;
 }
 
